app/src/lib/mouse-scroller.ts

```diff
@@ -0,0 +1,178 @@
+    }
+
+    this.updateMouseScroll(scrollable, mouseY)
+  }
+
+  /**
+   * The scrolling action is wrapped in a continual time out, it will
+   * continue to scroll until it reaches the end of the scroll area.
+   */
+  private updateMouseScroll(scrollable: Element, mouseY: number) {
+    window.clearTimeout(this.scrollTimer)
+
+    if (this.scrollVerticallyOnMouseNearEdge(scrollable, mouseY)) {
+      this.scrollTimer = window.setTimeout(() => {
+        this.updateMouseScroll(scrollable, mouseY)
+      }, 30)
+    }
+  }
+
+  /**
+   * Cleat the scroller's timeout.
+   */
+  public clearScrollTimer() {
+    window.clearTimeout(this.scrollTimer)
+  }
+
+  /**
+   * Method to scroll elements based on the mouse position. If the user moves
+   * their mouse near to the edge of the scrollable container, then, we want to
+   * invoke the scroll.
+   *
+   * Returns false if mouse is not positioned near the edge of the scroll area
+   * or the the scroll position is already at end of scroll area.
+   */
+  private scrollVerticallyOnMouseNearEdge(
+    scrollable: Element,
+    mouseY: number
+  ): boolean {
+    // how far away from the edge of container to invoke scroll
+    const { top, bottom } = scrollable.getBoundingClientRect()
+    const distanceFromBottom = bottom - mouseY
+    const distanceFromTop = mouseY - top
+
+    if (distanceFromBottom > 0 && distanceFromBottom < this.defaultScrollEdge) {
+      const scrollDistance = this.getScrollDistance(distanceFromBottom)
+      return this.scrollDown(scrollable, scrollDistance)
+    }
+
+    if (distanceFromTop > 0 && distanceFromTop < this.defaultScrollEdge) {
+      const scrollDistance = this.getScrollDistance(distanceFromTop)
+      return this.scrollUp(scrollable, scrollDistance)
+    }
+
+    return false
+  }
+
+  /**
+   * Calculate the scroll amount (which in turn is scroll speed). It uses the
+   * distance from the scroll edge to get faster as the user moves their mouse
+   * closer to the edge.
+   */
+  private getScrollDistance(distanceFromScrollEdge: number) {
+    const intensity = this.defaultScrollEdge / distanceFromScrollEdge
+    return this.scrollSpeed * intensity
+  }
+
+  /**
+   * Scrolls an element up by given scroll distance.
+   * Returns false if already at top limit else true.
+   */
+  private scrollUp(scrollable: Element, scrollDistance: number): boolean {
+    const limit = 0
+    if (scrollable.scrollTop === limit) {
+      return false
+    }
+
+    const inBounds = scrollable.scrollTop > scrollDistance
+    const scrollTo = inBounds ? scrollable.scrollTop - scrollDistance : limit
+    scrollable.scrollTo({ top: scrollTo })
+    return true
+  }
+
+  /**
+   * Scrolls an element up by given scroll distance.
+   * Returns false if already at bottom limit else true.
+   */
+  private scrollDown(scrollable: Element, scrollDistance: number): boolean {
+    const limit = scrollable.scrollHeight - scrollable.clientHeight
+    if (scrollable.scrollTop === limit) {
```

Comment:
Same as the other comment, feel free to ignore it! 😄 
```suggestion
    if (scrollable.scrollTop >= limit) {
```